File: elovate_backend/src/main/java/com/tuwien/elovate/dtos/statistics/LeaderboardPlayerDto.java
package com.tuwien.elovate.dtos.statistics;

/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */

public interface LeaderboardPlayerDto {
    Long getUserId();

    String getUsername();

    Long getGameId();

    double getRating();

    String getCountryCode();

    Long getLeaderboardPosition();
}

File: elovate_frontend/src/dtos/Statistics/LeaderboardPlayerDto.ts
/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */
export interface LeaderboardPlayerDto {
  userId: number;
  username: string;
  gameId: number;
  rating: number;
  countryCode?: string;
  leaderboardPosition?: number;
}
